src/validation.cpp

```diff
@@ -3542,6 +3542,80 @@ static void NotifyHeaderTip() LOCKS_EXCLUDED(cs_main) {
     }
 }
 
+bool IsAncestor(const CBlockIndex* pindexChain, const CBlockIndex* pindexCheck)
+{
+    if (!pindexChain || !pindexCheck)
+        return false;
+    if (pindexChain->nHeight <= pindexCheck->nHeight)
+        return false;
+
+    return pindexChain->GetAncestor(pindexCheck->nHeight)->GetBlockHash() == pindexCheck->GetBlockHash();
+}
+
+static const unsigned int PRUNE_DEPTH = 120;
```

Comment:
@mimirmim wondering if what I just found will address your underlying concern.  
we have `DEFAULT_MAX_REORG_DEPTH = 100`; and is settable with -maxreorg.  So I'm going to look at removing `PRUNE_DEPTH` and using nMaxReorgDepth in it's place.

Reply:
Yeah this is great. I am not seeing issues of unknown block headers of peers anymore.